src/core/tools/tool_line_to_track.cpp

```diff
@@ -0,0 +1,72 @@
+#include "tool_line_to_track.hpp"
+#include "board/board.hpp"
+#include "common/line.hpp"
+#include "imp/imp_interface.hpp"
+#include "util/selection_util.hpp"
+#include "util/uuid.hpp"
+#include <iostream>
+#include <functional>
+#include <vector>
+#include <unordered_set>
+#include "document/idocument_board.hpp"
+
+namespace horizon {
+
+bool ToolLineToTrack::can_begin()
+{
+    if (!(doc.r->has_object_type(ObjectType::LINE) && doc.r->has_object_type(ObjectType::JUNCTION)))
```

Comment:
Since this tool only makes sense on boards, this could be simplified to

```suggestion
    if (!doc.b)
```